test/suites/subnet/vnet_subnet_id_test.go

```diff
@@ -0,0 +1,206 @@
+
+		env.ExpectCreated(nodeClass, nodePool, dep)
+
+		env.EventuallyExpectCreatedNodeClaimCount("==", 1)
+		nodes := env.EventuallyExpectCreatedNodeCount("==", 1)
+		env.EventuallyExpectHealthyPodCount(selector, numPods)
+
+		vm := env.GetVM(nodes[0].Name)
+		Expect(vm.Properties).ToNot(BeNil())
+		Expect(vm.Properties.NetworkProfile).ToNot(BeNil())
+		Expect(vm.Properties.NetworkProfile.NetworkInterfaces).To(HaveLen(1))
+		Expect(vm.Properties.NetworkProfile.NetworkInterfaces[0].ID).ToNot(BeNil())
+		nicID, err := arm.ParseResourceID(*vm.Properties.NetworkProfile.NetworkInterfaces[0].ID)
+		Expect(err).ToNot(HaveOccurred())
+
+		// The NIC should have the right subnet
+		nic := env.GetNetworkInterface(nicID.Name)
+		Expect(nic.Properties).ToNot(BeNil())
+		Expect(nic.Properties.IPConfigurations).To(HaveLen(1))
+		Expect(nic.Properties.IPConfigurations[0].Properties).ToNot(BeNil())
+		Expect(nic.Properties.IPConfigurations[0].Properties.Subnet).ToNot(BeNil())
+		Expect(nic.Properties.IPConfigurations[0].Properties.Subnet.ID).To(Equal(subnet.ID))
+
+		// The NIC should have the right NSG
+		Expect(nic.Properties.NetworkSecurityGroup).ToNot(BeNil())
+		Expect(nic.Properties.NetworkSecurityGroup.ID).ToNot(BeNil())
+		Expect(*nic.Properties.NetworkSecurityGroup.ID).To(MatchRegexp(`aks-agentpool-\d{8}-nsg`))
+	})
+	It("should reject the AKSNodeClass if the subnet ID is invalid", func() {
+		nodeClass.Spec.VNETSubnetID = lo.ToPtr("/subnets/fake-subnet")
+		err := env.Client.Create(env.Context, nodeClass)
+		Expect(err).To(HaveOccurred())
+		Expect(err.Error()).To(ContainSubstring("Invalid value"))
+		Expect(err.Error()).To(ContainSubstring("spec.vnetSubnetID"))
+		Expect(err.Error()).To(ContainSubstring("should match"))
+	})
+
+	DescribeTable("should mark the AKSNodeClass as unready if the subnetID doesn't belong to the cluster vnet",
+		func(modifyComponents func(utils.VnetSubnetResource) utils.VnetSubnetResource) {
+			vnet := env.GetClusterVNET()
+			vnetID := lo.FromPtr(vnet.ID)
+
+			vnetResourceID, err := arm.ParseResourceID(vnetID)
+			Expect(err).ToNot(HaveOccurred())
+
+			baseComponents := utils.VnetSubnetResource{
+				SubscriptionID:    vnetResourceID.SubscriptionID,
+				ResourceGroupName: vnetResourceID.ResourceGroupName,
+				VNetName:          vnetResourceID.Name,
+				SubnetName:        "test-subnet",
+			}
+
+			modifiedComponents := modifyComponents(baseComponents)
+
+			subnetID := utils.GetSubnetResourceID(
+				modifiedComponents.SubscriptionID,
+				modifiedComponents.ResourceGroupName,
+				modifiedComponents.VNetName,
+				modifiedComponents.SubnetName,
+			)
+
+			nodeClass.Spec.VNETSubnetID = lo.ToPtr(subnetID)
+			env.ExpectCreated(nodeClass)
+
+			Eventually(func(g Gomega) {
+				g.Expect(env.Client.Get(env.Context, client.ObjectKeyFromObject(nodeClass), nodeClass)).To(Succeed())
+				condition := nodeClass.StatusConditions().Get(v1beta1.ConditionTypeSubnetsReady)
```

Comment:
nit: could check reason in both this test case, and the `should mark the AKSNodeClass as unready if the subnet is NotFound and fall back to a different nodeclass` test case